src/vanilla/store.ts

```diff
@@ -513,11 +544,14 @@ const buildStore = (getAtomState: StoreArgs[0]): Store => {
       } else {
         r = writeAtomState(pending, a, aState, ...args) as R
       }
-      flushPending(pending)
+      flushPending(pending, !isSync)
```

Comment:
This makes me feel like there's a bug in an edge case without this PR or we can improve something without this PR.

Reply:
I actually wonder if we can delete the synchronous call to flush pending.

Reply:
can you elaborate?

Reply:
Oh, do you mean like `if (!isSync) flushPending(pending)`? That's exactly my implication with the first comment.

Reply:
I mean if we remove synchronous flushPending from writeAtomState, then we should a flushPending after this line:
https://github.com/pmndrs/jotai/pull/2801/files#diff-849e7b1b832999901f65b02d61703f4e35f157d3f24387c13bf1e00e5a701dd3R617

Reply:
#2804 Let's focus on it separately. If you can help it, I appreciate it. Otherwise, I'll tackle when I have time.

Reply:
https://github.com/pmndrs/jotai/pull/2804 Let's focus on it separately. If you can help it, I appreciate it. Otherwise, I'll tackle when I have time.